spinalcordtoolbox/centerline/core.py

```diff
@@ -156,92 +114,140 @@ def get_centerline(im_seg, param=ParamCenterline(), verbose=1, remove_temp_files
+
+        elif param.algo_fitting == 'nurbs':
+            from spinalcordtoolbox.centerline.nurbs import b_spline_nurbs
+            point_number = 3000
+            # Interpolate such that the output centerline has the same length as z_ref
+            x_mean_interp, _ = curve_fitting.linear(z_mean, x_mean, z_ref, 0)
+            y_mean_interp, _ = curve_fitting.linear(z_mean, y_mean, z_ref, 0)
+            x_centerline_fit, y_centerline_fit, z_centerline_fit, x_centerline_deriv, y_centerline_deriv, \
+                z_centerline_deriv, error = b_spline_nurbs(x_mean_interp, y_mean_interp, z_ref, nbControl=None,
+                                                           point_number=point_number, all_slices=True)
+            # Normalize derivatives to z_deriv
+            x_centerline_deriv = x_centerline_deriv / z_centerline_deriv
+            y_centerline_deriv = y_centerline_deriv / z_centerline_deriv
+            fig_title = 'Algo={}, NumberPoints={}'.format(param.algo_fitting, point_number)
+
+        else:
+            logger.error('algo_fitting "' + param.algo_fitting + '" does not exist.')
+            raise ValueError
+
+        # Create an image with the centerline
+        im_centerline = im_seg.copy()
+        im_centerline.data = np.zeros(im_centerline.data.shape)
+        # Assign value=1 to centerline. Make sure to clip to avoid array overflow.
+        # TODO: check this round and clip-- suspicious
+        im_centerline.data[round_and_clip(x_centerline_fit, clip=[0, im_centerline.data.shape[0]]),
+                           round_and_clip(y_centerline_fit, clip=[0, im_centerline.data.shape[1]]),
+                           z_ref] = 1
+        # reorient centerline to native orientation
+        im_centerline.change_orientation(native_orientation)
+        im_seg.change_orientation(native_orientation)
+        # TODO: Reorient centerline in native orientation. For now, we output the array in RPI. Note that it is tricky to
+        #   reorient in native orientation, because the voxel center is not in the middle, but in the top corner, so this
+        #   needs to be taken into accound during reorientation. The code below does not work properly.
+        # # Get a permutation and inversion based on native orientation
+        # perm, inversion = _get_permutations(im_seg.orientation, native_orientation)
+        # # axes inversion (flip)
+        # # ctl = np.array([x_centerline_fit[::inversion[0]], y_centerline_fit[::inversion[1]], z_ref[::inversion[2]]])
+        # ctl = np.array([x_centerline_fit, y_centerline_fit, z_ref])
+        # ctl_deriv = np.array([x_centerline_deriv[::inversion[0]], y_centerline_deriv[::inversion[1]], np.ones_like(z_ref)])
+        # return im_centerline, \
+        #        np.array([ctl[perm[0]], ctl[perm[1]], ctl[perm[2]]]), \
+        #        np.array([ctl_deriv[perm[0]], ctl_deriv[perm[1]], ctl_deriv[perm[2]]])
```

Comment:
I'm wondering whether or not we should remove this dead code in the process.

The `TODO` is a bit confusing to me, because it implies we _don't_ reorient the centerline... when... we very clearly do, not 2 lines above?

I'm wondering whether the problem described in this comment is actually still a problem... :thinking:

Reply:
This file was born with this commented out code in the gigantic PR #2147 four years ago, but even in that PR I don't see any deleted code that it would correspond to. If this TODO points to a genuine bug, I don't think this comment is going to be the reason we find/fix it. Let's get rid of it!